File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    bool isVowel(char ch) {
    return (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' ||
            ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U');
    }

    int maxVowels(string s, int k) {
        int left = 0;
        int count = 0;
        int maxVowels = INT_MIN;
        for(int right = 0; right < s.length(); right++){
            if(isVowel(s[right])) count++;

            if(right - left + 1 == k){
                maxVowels = max(maxVowels, count);

                if(isVowel(s[left])) count--;

                left++;
            }
        }   
        return maxVowels;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def isVowel(self, ch):
        tmp = ch.lower()
        vowels = {'a', 'e', 'i', 'o', 'u'}
        if tmp in vowels:
            return True
        return False

    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        left = 0
        maxLen = float('-inf')
        count = 0

        for right in range(len(s)):
            if self.isVowel(s[right]):
                count += 1
            
            if right - left + 1 == k:
                maxLen = max(maxLen, count)
                if self.isVowel(s[left]): count -= 1
                left += 1
        
        return maxLen
        